app.rb

```diff
@@ -0,0 +1,179 @@
+require_relative './student'
+require_relative './book'
+require_relative './rental'
+require_relative './teacher'
+require_relative './classroom'
+
+# rubocop:disable Metrics
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @class = Classroom.new('Grade 5')
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+    puts
+    select
+  end
+
+  def select
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    option = gets.chomp
+    run_choice option
+  end
+
+  def run_choice(input)
+    case input
+    when '1'
+      list_books
+    when '2'
+      list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals
+    when '7'
+      puts 'See you again'
+    else
+      puts 'Invalid! Please enter a number from 1 to 7'
+      select
+    end
+  end
+
+  def list_books
+    puts 'No books added!' if @books.empty?
+
+    @books.each { |book| puts "Title: #{book.title}, author: #{book.author}" }
+    puts ''
+    select
+  end
+
+  def list_people
+    puts 'No person added!' if @people.empty?
+
+    @people.map { |person| puts "[#{person.class}] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}" }
+    puts ''
+    select
+  end
+
+  def create_person
+    print 'Select (1) to create Student and (2) to create Teacher: '
+    choice = gets.chomp
+    case choice
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      print 'Invalid input! Please select either 1  or 2: '
+    end
+  end
+
+  def create_student
+    print 'Enter age: '
+    age = gets.chomp.to_i
+
+    print 'Enter your name: '
+    name = gets.chomp
+
+    new_student = Student.new(@classroom, age, name)
+    @people.push(new_student)
+
+    puts "Student #{name.upcase} created successfully"
+
+    select
+  end
```

Comment:
- [ ] Kindly add "Has parent permission? [Y/N]" question to be similar to the 👉🏽  [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ)